plugin/evm/vm.go

```diff
@@ -807,9 +807,8 @@ func (vm *VM) preBatchOnFinalizeAndAssemble(header *types.Header, state *state.S
 		// Note: snapshot is taken inside the loop because you cannot revert to the same snapshot more than
 		// once.
 		snapshot := state.Snapshot()
-		rules := vm.chainConfig.Rules(header.Number, params.IsMergeTODO, header.Time)
-		rulesExtra := params.GetRulesExtra(rules)
-		if err := vm.verifyTx(tx, header.ParentHash, header.BaseFee, state, *rulesExtra); err != nil {
+		rules := params.GetRulesExtra(vm.chainConfig.Rules(header.Number, params.IsMergeTODO, header.Time))
```

Comment:
```suggestion
		ethrules := vm.chainConfig.Rules(header.Number, params.IsMergeTODO, header.Time)
		rules := params.GetRulesExtra(ethrules)
```

I agree with keeping the variable actually being used as named `rules` but think the line is a bit unwieldy as it is.

EDIT: I see that this is a common occurrence so, alternatively, you could abstract the pattern as a method like `rules := vm.rulesExtra(header)`.

Reply:
Did it with function

```go
func (vm *VM) rulesExtra(number *big.Int, time uint64) extras.Rules {
	ethrules := vm.chainConfig.Rules(number, params.IsMergeTODO, time)
	return *params.GetRulesExtra(ethrules)
}
```

So number+time instead of header argument, such that it can be used with blocks without having to deep copy the header.